### ISSUE TITLE
The template Home - Particles 5.5.12 displays does not display mainmenu

### ISSUE THREAD

**Issue by kosh2323** [gantry/gantry5#3022]
For some reason, the mainmenu menu is not displayed on the main page, and I can't figure it out. I make a separate Home - Particles template for the main page, I set the Assignments switch only for the main page, as a result, the menu is not displayed on the main page, everything is fine on all the others. Even if you make a duplicate from the Default template, and assign Default(2) only for the main page, then everything is correct and normal. Can someone tell me what the problem might be, I didn't find it in the official documentation.
![Home Partical1](https://user-images.githubusercontent.com/82117170/161259720-b5756652-e42c-42be-b473-588871b74823.jpg)

![Home Partical2](https://user-images.githubusercontent.com/82117170/161259666-9c2a4afc-8efd-43c2-9cc7-eab8b277be4b.jpg)

![Home Partical3](https://user-images.githubusercontent.com/82117170/161259691-325602a8-6327-4a8e-9bf2-f307754a286c.jpg)

![Home Partical4](https://user-images.githubusercontent.com/82117170/161259784-74516795-54c4-4752-985f-e7af034ba95e.jpg)

![Home Partical5](https://user-images.githubusercontent.com/82117170/161259803-d1791ed2-d8eb-4324-b334-c67193934375.jpg)

![Default1](https://user-images.githubusercontent.com/82117170/161259857-42b5003a-9398-42ac-bc4e-a71313b476dd.jpg)

![Double1](https://user-images.githubusercontent.com/82117170/161259894-fc31b861-ca3d-4031-a5f6-2e4ee7a7e933.jpg)

![Double2](https://user-images.githubusercontent.com/82117170/161259914-8eab067d-e6cf-471d-9ae6-8a25448d6070.jpg)

![Double3](https://user-images.githubusercontent.com/82117170/161259932-add9637a-9eea-40fa-aa7f-a211c2c5db61.jpg)

![Double4](https://user-images.githubusercontent.com/82117170/161259956-f113585a-cddb-45a3-894c-8ee0e048c10b.jpg)

**Comment by kosh2323** [comment]
Strange behavior of the Home - Particles template, if none of the modules are enabled in the Header section, then the mainmenu is not displayed, because the mainmenu itself is in another Navigation section. This is not present in the Default template. Is this a feature or a template bug?

![Home - Particles6](https://user-images.githubusercontent.com/82117170/161382874-f9e707a0-194c-4f6c-b5a6-cdb570e28823.jpg)

**Comment by mahagr** [comment]
This may be caused by some extensions which break the layout assignments by calling certain Joomla API methods too early. 

Basically, it always uses the default layout, even in Joomla's own template.

You should look into System plugins and try disabling some to figure out which extension does this.

**Comment by kosh2323** [comment]
> This may be caused by some extensions which break the layout assignments by calling certain Joomla API methods too early.
> 
> Basically, it always uses the default layout, even in Joomla's own template.
> 
> You should look into System plugins and try disabling some to figure out which extension does this.

Thanks for the answer Matias, I'll try to figure it out, although it will be easier to make a duplicate of the default template and assign it to one page. I checked, - it works without problems.

**Comment by mahagr** [comment]
Hmm, then I'm not sure what is going on. Which template are you using when getting the issue? I may want to try it myself.

**Comment by kosh2323** [comment]
> Hmm, then I'm not sure what is going on. Which template are you using when getting the issue? I may want to try it myself.

Yes, it seems I understood what was going on, the default link fonts are white, and when I removed Owl Carousel from the Header section, there was a white background and no menu links were visible on it, sorry for my mistake and for taking your time. Thanks.

**Comment by kosh2323** [comment]
There are problems with assigning the entire background section and font colors. In this example, I copied the default template and named it by default 2. I have to assign the background of the section through separate classes.
![1](https://user-images.githubusercontent.com/82117170/162449720-111af5a8-c520-4e4d-85de-93ee4e690258.jpg)
![2](https://user-images.githubusercontent.com/82117170/162449740-e2088277-5e01-4604-9303-e7afe7bb1456.jpg)

**Comment by marktaylor46** [comment]
I checked this and it works fine for me on my helium site. So... I suggest the following.
1. Check what outline is ACTUALLY being used on the frontend to render that page - there is a class on the body tag of the form outline-123 (where "123" denotes the outline ID being used).
2. Go to Gantry 5 admin and click on the "outlines" button and that will show you all your outline (with their IDs) so you can check that the outline you have changed is the outline being used by that page.
3. Clear browser and server cache
4. If the first two items check out ok then use a browser inspector tool to see what statement is setting the background-color for that section (is is a theme CSS file or is it coming from somewhere else).

post a link to your site too if you still have issues.

Regards, Mark.

**Comment by kosh2323** [comment]
Hi Mark, thanks for wanting to help.
And so I take the default template 2, it has ID=14, it has a Feature section, it is not inherited from Base Outline, in this section I set the font color and background, click save and recompile, mainmenu is located in this section. 

![3](https://user-images.githubusercontent.com/82117170/162474691-a6de5010-c676-474c-a69d-abbb324d9789.jpg)

![4](https://user-images.githubusercontent.com/82117170/162474727-97f973df-c442-4c3d-b453-6c1f7dabbcc9.jpg)


And I see that the font color and background have not changed. I look at the code, and I see that the section has not been assigned a style, it is also not in CSS. There is no style in nested elements either. 

![5](https://user-images.githubusercontent.com/82117170/162475628-385f5885-58ad-42b9-bade-f364788a7b3c.jpg)


![6](https://user-images.githubusercontent.com/82117170/162475658-d6e01368-ccd6-495f-882a-8e3904280abe.jpg)


Why doesn 't it change ?Maybe the mainmenu module itself somehow affected, then I will delete the menu in this section and place a simple html code, the results are the same, the background and font color have not changed.

![7](https://user-images.githubusercontent.com/82117170/162475780-0a3d1257-ffa6-4ade-8c05-87f7d65e1796.jpg)


![8](https://user-images.githubusercontent.com/82117170/162475797-0e82803a-913c-4777-9d34-87d66525c73d.jpg)

**Comment by marktaylor46** [comment]
Send me login details and I will check for you.

Regards, Mark.

**Comment by kosh2323** [comment]
> Send me login details and I will check for you.

I can't, I do it locally. The only thing that suspicion falls on is that the layout itself for Default 2 clicked the download button and selected the Load Gantry4 model. It is very lacking here to add the necessary sections yourself, at least from the possibly available ones.

**Comment by marktaylor46** [comment]
Send me an akeeba backup file of the site then instead.

Regards, Mark.

**Comment by kosh2323** [comment]
> Send me an akeeba backup file of the site then instead.
> 
> Regards, Mark.

I don't use Akeeba, I just import the database and files, if you can, then the link is here

_https://drive.google.com/file/d/1yzKIAef7rvasPPMFzYBMnzRZMtbtRzuZ/view?usp=sharing

**Comment by kosh2323** [comment]
Forgot to write user, database, password, administrator:   test2
pass admin: test2test2test2

**Comment by marktaylor46** [comment]
Ok, so I loaded this locally and it's obvious what the issue is.

The "default 2" outline layout seems to have been manually manipulated (which you should not do). The "features" section has had it's name changed to "feature" (singular) and this is why the features.scss file is not applying to your new section.

You should not manually manipulate layouts of existing outlines. What you should do instead is to create new layout preset and then load this new preset on your outline using the "LOAD" button on the layout.

https://docs.gantry.org/gantry5/advanced/creating-layout-presets

When creating new layout presets you need to ensure that you also add and load any other related files and settings too (e.g. scss files, setting yaml files, style files, preset files, etc etc).

So, in summary it would appear that your manual manipulation of the outline yaml file caused this issue (when you renamed the section).

I hope that helps.

Regards, Mark.

**Comment by kosh2323** [comment]
> Ok, so I loaded this locally and it's obvious what the issue is.
> 
> The "default 2" outline layout seems to have been manually manipulated (which you should not do). The "features" section has had it's name changed to "feature" (singular) and this is why the features.scss file is not applying to your new section.
> 
> You should not manually manipulate layouts of existing outlines. What you should do instead is to create new layout preset and then load this new preset on your outline using the "LOAD" button on the layout.
> 
> https://docs.gantry.org/gantry5/advanced/creating-layout-presets
> 
> When creating new layout presets you need to ensure that you also add and load any other related files and settings too (e.g. scss files, setting yaml files, style files, preset files, etc etc).
> 
> So, in summary it would appear that your manual manipulation of the outline yaml file caused this issue (when you renamed the section).
> 
> I hope that helps.
> 
> Regards, Mark.

Thanks for the answer and help Mark. You have helped me a lot to understand and understand. In the beginning, I wanted to do this, as it is written in the documentation, but there are some opinions that using templates via custom, this leads to a slowdown in the site. So I decided to try another option for now. Copy from Outline Default and do Default2 and work with it, well, apparently this caused problems, because there are no separate YAML and SCSS for the new copy, I didn't know that. In addition, it turns out that if you upload a ready-made layout, for example Gantry 4, it looks like it either can't be changed in any way, or it needs to create its own YAML and Scss styles, so it turns out? Or again I don't quite understand it?

P.S. I just used to do it in the Astroid template, where sections can be created, moved, deleted. This is certainly not enough here.

**Comment by marktaylor46** [comment]
Huh? Every single outline on your site is going to be loaded from custom - because that's where all your configuration changes to the layout are stored - in /templates/g5_helium/custom/config 

When you use a layout preset (and there are many available) it copies the structure to your own outline layout copy (in the same location given above) - then does not use the preset again at all. If you manually choose to alter the config copy of the layout yaml then if you were to change layout preset or delete the outline (for whatever reason) then you would lose all your modifications to the layout structure. Hence why it is better to create a new layout preset (which won't ever get overwritten).

You could change the existing layout presets (but I don't recommend that you do that as they will be overwritten again by template updates)... better to take a copy of an existing preset and rename it (which won't get overwritten). 

The rules still apply though... if you start renaming sections or adding new ones then there is much more to do too (create scss, creating styles presets, creating setting yaml, etc etc).

So, I don't know who is telling you that using templates from custom slows down your site - but it's not true - all your outlines will come from custom anyway.

Regards, Mark.

**Comment by kosh2323** [comment]
> Huh? Every single outline on your site is going to be loaded from custom - because that's where all your configuration changes to the layout are stored - in /templates/g5_helium/custom/config
> 
> When you use a layout preset (and there are many available) it copies the structure to your own outline layout copy (in the same location given above) - then does not use the preset again at all. If you manually choose to alter the config copy of the layout yaml then if you were to change layout preset or delete the outline (for whatever reason) then you would lose all your modifications to the layout structure. Hence why it is better to create a new layout preset (which won't ever get overwritten).
> 
> You could change the existing layout presets (but I don't recommend that you do that as they will be overwritten again by template updates)... better to take a copy of an existing preset and rename it (which won't get overwritten).
> 
> The rules still apply though... if you start renaming sections or adding new ones then there is much more to do too (create scss, creating styles presets, creating setting yaml, etc etc).
> 
> So, I don't know who is telling you that using templates from custom slows down your site - but it's not true - all your outlines will come from custom anyway.
> 
> Regards, Mark.

Hi Mark. Thank you again for the answer and recommendations. I understood some things, but there are nuances that I don't understand yet. Let's say that the layout that is now initially in Base Outline suits me, and I would like it to be the same in Base Outline, Default, Home Partial. Then the Navigation, Header, Footer sections would be inherited from the Base Outline. And in the Default and templates
Home Partial would only change positions and modules. How should this scheme be done?

Will it be enough for me to copy these layouts or create all new ones? Or copy the Base Outline 2 times, rename the 2 copied ones and leave the base one? I don't want to do it through YAML.

**Comment by marktaylor46** [comment]
The base outline layout (in joomla) in itself does nothing - you can't assign it to any of your pages - but this is the place to set up all the "common" parts of your pages for your site (certainly your navigation, menu, logo, footer, copyright , etc). If you really want to use that one layout for the whole of your site (which I find puzzling) then you would also set up "module position" particles on the layout of the outline (into which you would assign modules in joomla module manager).

If all the pages of your site are going to use that layout then the only other outline that you would need would be "default" (which you would make the default template in template manager). Then, on the layout of the "default" outline your would tell every single section to "inherit" it's content from the  base outline layout. You can use this one default outline for all of the pages of your site because their is nothing page-specific on it's layout. You differentiate the pages of your site by assigning different modules into the module positions.

Conversely, the other approach you can take is to create (for example) a "home" outline and have it inherit the "common" parts of your page into it's layout, then instead of module positions on the layout your could just put the particles required on that layout. You can repeat this for other pages of your site too. But, do you see that in doing this you are making each outline page-specific - so you're going to be creating an outline for each page this way.

In reality you will use a mixture of both techniques.

The other thing to be aware of is that a layout does not have to be the same "shape". Normally your home page would look different to the internal pages of your site - for this reason it would use a different layout preset to the "default" outline (which should use the "default" preset). If you click on the "LOAD" button on the layout of any outline you will see that their are many different layout preset that can be used for an outline (with sidebar, two sidebars, three sidebar etc etc). Whichever layout preset you elect to use you can still "inherit" content from another outline (usually base outline) even if the layouts are using different presets.

Don't "copy" base outline - just create a new outline, pick the preset you want and save, then on the layout choose to "inherit" whatever the "common" parts of your page are from the base outline.

It's a shame you're not using a RocketTheme template because I think it would be a lot clearer to you if you looked at demo examples of RT sites. RT provides a rocketlauncher package that provides an exact replica of the demo so you could see how everything was done.

I think I've given you more than enough to go on here - Sorry but I can't provide full training on the use of Gantry 5 in this github ticket.

Regards, Mark.

**Comment by kosh2323** [comment]
Yes, Mark, thank you very much, you have already given me a lot of material and spent a lot of your time on me. Excuse me if my questions were stupid and banal. I'll try to figure it out now to the end. Success.

**Comment by kosh2323** [comment]
Sorry Mark, the last thing I want to say, for the sake of interest, I tried another template Hydrogen 5.5.12. And it works there, the background and colors are correctly inherited from the Base Outline. In Helium, the background is not inherited in Home - Particles.

**Comment by kosh2323** [comment]
> Hello @kosh2323,
> 
> I've only read through part of this ticket so please excuse this information if it doesn't help but I believe the issue you are experiencing has to do with the styling "classes" that are added specifically for the Helium template. I faced this issue in the past when I started using it.
> 
> The class in question is added to the helium template under "Page Settings" >> "Body Attributes" >> "Body Classes" and there you will find **g-helium-style**.
> 
> I've added a picture so you can see how it has been added to the template. ![image](https://user-images.githubusercontent.com/12389693/162691308-0fe1e337-4df8-4675-b44f-81cb6bde19c4.png)
> 
> If you would like to modify the styling, you can have a look at the file located here: **/templates/g5_helium/scss/helium/styles/_helium-style.scss** Then you will need to create your changes in the **custom.scss** file.
> 
> Good luck and I hope this has helped.

Hi Michael. Thanks for the answer, and thanks a lot for the help. Indeed, this clarified the situation, these styles prevented inheriting the background. Why it was necessary for developers to confuse so, it is unclear. In addition, nothing is said about this in the documentation.

**Comment by marktaylor46** [comment]
@kosh2323 I explained to you why it did not work on Helium already - please re-read what I said before. Basically you had changed the name of the section in the outline from "features" to "feature" which meant that the "features.scss" file would not apply to that new section name and that is why no change occurred when you edited the "features" styling.

I have checked the Helium files (as provided) and the section is called "features" throughout allthe outlines and layout presets. and the scss file is correctly named as "features" too and does contain scss that changes the "features" section.

In Hydrogen (a completely different template) then the section is called "feature", the scss file is called "feature" and all the outlines and layout preset refer to "feature" - so again everything is correct.

So, in conclusion there is nothing wrong with either Helium or Hydrogen. It is your renaming of a section in the outline yaml file that caused your issue.

Regards, Mark.

**Comment by kosh2323** [comment]
> @kosh2323 I explained to you why it did not work on Helium already - please re-read what I said before. Basically you had changed the name of the section in the outline from "features" to "feature" which meant that the "features.scss" file would not apply to that new section name and that is why no change occurred when you edited the "features" styling.
> 
> I have checked the Helium files (as provided) and the section is called "features" throughout allthe outlines and layout presets. and the scss file is correctly named as "features" too and does contain scss that changes the "features" section.
> 
> In Hydrogen (a completely different template) then the section is called "feature", the scss file is called "feature" and all the outlines and layout preset refer to "feature" - so again everything is correct.
> 
> So, in conclusion there is nothing wrong with either Helium or Hydrogen. It is your renaming of a section in the outline yaml file that caused your issue.
> 
> Regards, Mark.

Yes, Mark, I understood this the first time, but yesterday I tried a clean installation of Joomla in 4.12 and Helium 5.5.12. I didn't change anything at all, I just set the background of the Navigation section in the Base Outline and looked at how it is inherited in the Home - Particles and Default template and I saw that the background is not inherited only in Home - Particles, but as soon as I removed the style "g-helium-style" in Page Settings, everything became normal. Thanks.

**Comment by marktaylor46** [comment]
Ok well that is a different issue to that which you first mentioned. Yes there are demo classes that will override some settings made on the styles tab. Demo classes should be removed when you make your own site. "g-helium-styles" is one of them "g-logo" is another on the logo/image particle. These classes are just to make the demo's work only.

Regards, Mark.

**Comment by kosh2323** [comment]
> Ok well that is a different issue to that which you first mentioned. Yes there are demo classes that will override some settings made on the styles tab. Demo classes should be removed when you make your own site. "g-helium-styles" is one of them "g-logo" is another on the logo/image particle. These classes are just to make the demo's work only.
> 
> Regards, Mark.

Yes, everything is mixed up, but it's good that we still managed to find out everything. Thank you all, especially you Mark and N8Solutions.

**Comment by marktaylor46** [comment]
Please close the ticket now then - cheers.

You're very welcome.

Regards, Mark.
